internalshared/configutil/config.go

```diff
@@ -75,10 +76,12 @@ func ParseConfig(d string) (*SharedConfig, error) {
 	}
 
 	if result.DisableMlockRaw != nil {
-		if result.DisableMlock, err = parseutil.ParseBool(result.DisableMlockRaw); err != nil {
-			return nil, err
+		isDisabled, err := parseutil.ParseBool(result.DisableMlockRaw)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing 'disable_mlock': %w", err)
+		} else if !isDisabled {
+			return nil, fmt.Errorf("OpenBao does not support mlock; please disable and use CGroups instead")
```

Comment:
Small nitpick with the error message: CGroups are a Linux only feature, but mlock is also supported by the various BSDs we build binaries for. In light of this I'd rather ask the user to disable swap on their system